packages/material-ui/src/MenuItem/MenuItem.d.ts

```diff
@@ -1,33 +1,58 @@
 import { SxProps } from '@material-ui/system';
 import { Theme } from '@material-ui/core/styles';
-import { DistributiveOmit } from '@material-ui/types';
-import { ListItemTypeMap, ListItemProps } from '../ListItem';
-import { OverridableComponent, OverrideProps } from '../OverridableComponent';
-import { ExtendButtonBase } from '../ButtonBase';
+import { ExtendButtonBase, ExtendButtonBaseTypeMap } from '../ButtonBase';
+import { OverrideProps } from '../OverridableComponent';
 import { MenuItemClasses } from './menuItemClasses';
 
-export interface MenuItemTypeMap<P = {}, D extends React.ElementType = 'li'> {
-  props: P &
-    DistributiveOmit<ListItemTypeMap<P, D>['props'], 'children' | 'classes'> & {
-      /**
-       * The content of the component.
-       */
-      children?: React.ReactNode;
-      /**
-       * Override or extend the styles applied to the component.
-       */
-      classes?: Partial<MenuItemClasses>;
-      /**
-       * `classes` prop applied to the [`ListItem`](/api/list-item/) element.
-       */
-      ListItemClasses?: ListItemProps['classes'];
-      /**
-       * The system prop that allows defining system overrides as well as additional CSS styles.
-       */
-      sx?: SxProps<Theme>;
-    };
+export type MenuItemTypeMap<P = {}, D extends React.ElementType = 'li'> = ExtendButtonBaseTypeMap<{
+  props: P & {
+    /**
+     * If `true`, the list item is focused during the first mount.
+     * Focus will also be triggered if the value changes from false to true.
+     * @default false
+     */
+    autoFocus?: boolean;
+    /**
+     * The content of the component.
+     */
+    children?: React.ReactNode;
+    /**
+     * Override or extend the styles applied to the component.
+     */
+    classes?: Partial<MenuItemClasses>;
+    /**
+     * If `true`, compact vertical padding designed for keyboard and mouse input is used.
+     * The prop defaults to the value inherited from the parent Menu component.
+     * @default false
+     */
+    dense?: boolean;
+    /**
+     * If `true`, the component is disabled.
+     * @default false
+     */
+    disabled?: boolean;
+    /**
+     * If `true`, the left and right padding is removed.
+     * @default false
+     */
+    disableGutters?: boolean;
+    /**
+     * If `true`, a 1px light border is added to the bottom of the menu item.
+     * @default false
+     */
+    divider?: boolean;
+    /**
+     * Use to apply selected styling.
+     * @default false
+     */
+    selected?: boolean;
+    /**
+     * The system prop that allows defining system overrides as well as additional CSS styles.
+     */
+    sx?: SxProps<Theme>;
```

Comment:
```suggestion
```

Same as previous?

Reply:
If removed, it is all gone from docs api and proptype. Let's keep it for now.

Reply:
I think on the docs there is info that it extends some interface, so that should be enough.

Reply:
But `Button`, `IconButton` and `ListItemButton` also has its own `sx`, should we keep the same for `MenuItem`? not really sure.

Reply:
Let’s keep it, but leave this conversation open, so that we can address it later.